nifi-nar-bundles/nifi-hbase-bundle/nifi-hbase-processors/src/main/java/org/apache/nifi/hbase/PutHBaseRecord.java

```diff
@@ -295,13 +302,21 @@ protected PutFlowFile createPut(ProcessContext context, Record record, RecordSch
                     if (name.equals(rowFieldName)) {
                         continue;
                     }
-                    columns.add(new PutColumn(fam, clientService.toBytes(name), asBytes(name, schema.getField(name).get().getDataType().getFieldType(), record, asString)));
+                    columns.add(new PutColumn(fam, clientService.toBytes(name), asBytes(name,
+                            schema.getField(name).get().getDataType().getFieldType(), record, asString, complexFieldStrategy)));
+                }
+                String rowIdValue = record.getAsString(rowFieldName);
+                if (rowIdValue == null) {
+                    throw new Exception(String.format("Row ID was null for flowfile with ID %s", flowFile.getAttribute("uuid")));
                 }
-                retVal = new PutFlowFile(tableName, clientService.toBytes(record.getAsString(rowFieldName)), columns, flowFile);
+                byte[] rowId =  getRow(rowIdValue, fieldEncodingStrategy);
```

Comment:
Should this be the "row encoding strategy" rather than the "field encoding strategy"? 

There's actually separate properties for these (FIELD_ENCODING_STRATEGY vs. ROW_ID_ENCODING_STRATEGY).